src/olympia/versions/models.py

```diff
@@ -130,6 +134,10 @@ def from_upload(cls, upload, addon, platforms, channel, send_signal=True,
                     source=None, is_beta=False):
         from olympia.addons.models import AddonFeatureCompatibility
 
+        if addon.status == amo.STATUS_DISABLED:
+            raise VersionCreateError(
+                'Addon is Mozilla Disabled; no new versions are allowed.')
+
```

Comment:
The user shouldn't get to this point - the button is hidden; view url is a 404; and the API has a separate check before the method is called.  I was going to take it out but forgot - is it worth keeping as a fallback?